experimental/single_number_formatter/src/ule.rs

```diff
@@ -116,6 +127,20 @@ impl AsULE for CurrencyPatterns {
         let short_place_holder_index = ((short_prefix as u16) << 8) | second_byte as u16;
         let narrow_place_holder_index = ((narrow_prefix as u16) << 8) | third_byte as u16;
 
+        let short_place_holder_index = match short_place_holder_index {
+            NO_PLACE_HOLDER => None,
+            USE_ISO_CODE => Some(PlaceHolder::ISO),
+            index if index <= MAX_PLACE_HOLDER_INDEX => Some(PlaceHolder::Index(index)),
+            _ => panic!("Invalid short_place_holder_index"),
```

Comment:
Don't panic, make `index <= MAX_PLACE_HOLDER_INDEX` an invariant on `PlaceHolder` and uphold it everywhere.